### DISCUSSION TITLE
Bluepill is now an MNF project!

### DISCUSSION THREAD

**Discussion by ob** [MobileNativeFoundation/discussions#112] [General]
As some of you probably already know, LinkedIn has donated [Bluepill](https://github.com/MobileNativeFoundation/bluepill), a parallel iOS Simulator runner, to the MNF. See [the blog post](https://engineering.linkedin.com/blog/2021/releasing-bluepill-to-the-mobile-native-foundation) for details.

I want to expand a little behind some of the motivation for the donation, our goals, and how you can get involved. Bluepill was developed between 2016 and 2017 as a solution to the long times our CI runs were taking. Apple's test runner didn't support parallelism and running the tests in parallel made a tremendous impact in our CI build times.

As time passed, Bluepill gained other useful features. We discovered a lot of our tests were flaky and some developers were getting frustrated when their CI runs would fail due to tests that had nothing to do with the change they were making; so we added a retry mechanism and a way to flag tests as flaky. We found that Simulators sometimes crashed and the whole CI execution would grind to a halt; so we added retries for Simulator crashes too. We found that under some conditions tests would hang the app; so we added timeouts and the ability to get stack traces. We also incorporated Chrome's [trace profiling](https://www.chromium.org/developers/how-tos/trace-event-profiling-tool) into Bluepill, and JUnit support so that we could improve the performance of our tests and get test results in an easy to digest way.

Over the years, Apple has not been idle; they have added some of Bluepill's original capabilities to their own test runner; while some others are still missing. At the same time, Bluepill has become popular outside of LinkedIn; as other companies find value in using it in their own CI systems.

With the formation of the Mobile Native Foundation, we saw an opportunity to allow the iOS development community to determine what the best direction for the project should be. We are committed to working with the community and together figuring out what direction the project should take. I have created a [#bluepill channel](https://mobilenativefdn.slack.com/archives/C024BV0603X) in the [MNF Slack Channel](https://mobilenativefdn.slack.com), please get involved and help us make iOS Testing at scale great!

**Comment by keith** [comment]
Thanks @ob, I'm excited that bluepill is part of the MNF!

**Comment by piannaf** [comment]
First: Awesome! This is exciting!

Second: I noticed the link to the slack channel is malformed

**Reply by ob** [reply]
Fixed! Thanks for reporting it.

**Comment by bharathns** [comment]
Thank you @ob
